src/logger/cloudwatch.logger.ts

```diff
@@ -0,0 +1,91 @@
+import AWS from "aws-sdk"
+import Bluebird from "bluebird"
+import winston from "winston"
+import WinstonCloudwatch from "winston-cloudwatch"
+
+import { consoleLogger } from "./console.logger"
+
+// AWS
+const AWS_REGION_NAME = "ap-southeast-1"
+AWS.config.update({ region: AWS_REGION_NAME })
+const awsMetadata = new AWS.MetadataService()
+const metadataRequest = Bluebird.promisify<string, string>(
+  awsMetadata.request.bind(awsMetadata)
+)
+
+// Constants
+// TODO: Check this env var as it is not in example
+const LOG_GROUP_NAME = `${process.env.AWS_BACKEND_EB_ENV_NAME}/nodejs.log`
+
+// Retrieve EC2 instance since that is the cloudwatch log stream name
+async function getEc2InstanceId(): Promise<string> {
+  let id: string
+  try {
+    id = await metadataRequest("/latest/meta-data/instance-id").timeout(1000)
```

Comment:
[Perf] hmm i think we should env var this. worried that this would introduce a network call for every log that we make, which seems expensive 

Also to check in, this `id` wont change for a specific env right?

Reply:
this is actually legacy code that i preserved for the sake of compat - this function is also ever called in the constructor so it doesn't cause a perf hit. i didn't change the `timeout` because i didn't know if there was any additional effects - it was already there in the [original PR](https://github.com/isomerpages/isomercms-backend/pull/69/files) and there wasn't any comments wrt it .__.

Reply:
arh ok if just in constructor should be fine bah, i thought was per every `logger.<logger-type>()` call